src/components/CharacterSheet.js

```diff
@@ -1,38 +1,77 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import firebase from 'firebase';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
+import { Fab } from '@material-ui/core';
 import Profile from './Profile';
 import Attributes from './Attributes';
 import Skills from './Skills';
+import { TopAnchor } from './CustomStyled';
+import dungeonService from '../services/dungeonService';
 
 
-const CharacterSheet = ({ character }) => (
-  <SheetContainer>
-    <ProfileArea>
-      <Profile character={character} />
-    </ProfileArea>
-    <StatsArea>
-      <Attributes character={character} />
-    </StatsArea>
-    <SkillsArea>
-      <Skills character={character} />
-    </SkillsArea>
-    <WeaponsArea />
-    <EquipmentArea />
-  </SheetContainer>
-);
+const CharacterSheet = ({ data }) => {
+  const [character, updateCharacter] = useState(data);
+  const [isDirty, setDirty] = useState(false);
```

Comment:
What is Dirty?

Reply:
you familiar the term 'dirty form'? Basically if a form has been edited since the last save, its considered 'dirty'

Reply:
lol i am now 😂